packages/transaction-controller/src/TransactionController.ts

```diff
@@ -944,6 +945,33 @@ export class TransactionController extends BaseController<
     this.update({ transactions: this.trimTransactionsForState(transactions) });
   }
 
+  /**
+   * Add params to transactionMeta.
+   *
+   * @param transactionMetaId - ID of the transaction.
+   * @param securityAlertResponse - New params to be added to tarnsaction meta.
+   */
+  updateSecurityAlertResponse(
+    transactionMetaId: string,
+    securityAlertResponse: SecurityAlertResponse,
+  ) {
+    if (!transactionMetaId || !securityAlertResponse) {
+      return;
+    }
+    const { transactions } = this.state;
+    const index = this.state.transactions.findIndex(
```

Comment:
Rather than finding this manually, should we re-use the pattern from the other `updateX` methods and use `getTransaction`?

```
const transactionMeta = this.getTransaction(transactionId);
```